src/main.py

```diff
@@ -524,6 +712,16 @@ async def main() -> None:
             )
             deserializer = Deserializer(registry=registry_api)
 
+            # Redis client setup with retry and exponential backoff
+            # https://redis.io/kb/doc/22wxq63j93/how-to-manage-client-reconnections-in-case-of-errors-with-redis-py
+            redis_client = redis.Redis(host=redis_host,
+                                       retry=Retry(ExponentialBackoff(cap=redis_retry_delay_cap,
+                                                                      base=redis_retry_initial_delay),
+                                                   redis_retry_count),
+                                       retry_on_error=[ConnectionError, TimeoutError, ConnectionResetError],
+                                       health_check_interval=redis_health_check_interval)
+            await redis_client.aclose()
```

Comment:
Why are both `httpx.AsyncClient` and `redis.Redis` set up when only one is used? That means we could end up in a situation where an httpx failure prevents a Keda service from running, or vice versa.

I think it would make more sense to branch on `platform` higher up, implement the branches as separate functions, and factor out any shared code.

Reply:
`httpx.AsyncClient` is needed because it used by the Kafka Schema Registry client to [deserialize the next visit](https://github.com/lsst-dm/next_visit_fan_out/blob/14cb83c4a940bd243a21dc26697c9a81e86b393e/src/main.py#L711) message.

Reply:
Ok. I still think `main` could be factored a bit, but that's less critical.